tests/parametric/test_headers_tracecontext.py

```diff
@@ -774,22 +773,21 @@ def test_tracestate_w3c_p_extract(self, test_agent, test_library):
         assert case1["meta"]["_dd.parent_id"] == "0123456789abcdef"
 
         assert case2["name"] == "p_invalid"
-        assert case2["meta"]["_dd.parent_id"] == "XX!X"
+        assert "_dd.parent_id" not in case2["meta"]
```

Comment:
```suggestion
        assert case2["meta"]["_dd.parent_id"] == "0000000000000000"
```

Good catch here. Invalid parent_ids should not be propagated. `_dd.parent_id` should be set to `0000000000000000`.

Reply:
I might had overlooked the RFC. If `p` is invalid, why `_dd.parent_id` should be set to "0000000000000000"?

Reply:
Validation of the propagated `p` sub-key on extraction wasn't considered to be important - when it gets to the backend it'll need to be validated there regardless of whether we validated it or not as we don't have exclusive control over the contents of `tracestate`.

But fine with me either way, if validation whenever it gets extracted makes more sense then 👍

Reply:
@bouwkast not sure to understand. In scenario 2, the RFC spec that `_dd.parent_id` should be `0000000000000000`. Do you mean the backend will still have to validate `p`, thus `_dd.parent-id=0000000000000000` or `_dd.parent_id` is equivalent? If yes, then I think not sending `_dd.parent_id` is better resource wise.

Reply:
Thank you @mabdinur.

I think the confusion comes from:

> The RFC also states that when an **_invalid_** `p` value is extracted `_dd.parent_id` should be set to `0000000000000000`.

To know if `p` is invalid, we must run some validation process on `p`. 

My understanding of that _validation_ was:
> - The p sub-key MUST be injected into the dd list-member of the tracestate.
> - The value of p MUST be a 16-character lowercase hexadecimal encoded string.
> - The value of p MUST be set to the ID of the currently active span.
> - The value of p MUST be a non-zero value (0000000000000000 is invalid).
> - A : (colon) character MUST be placed between the key and the value. For example, p:0123456789abcdef

While the rest of the team understanding was (I guess):
> - The initial value MUST be a hexadecimal string 16 characters in length.

In the end, that's a misinterpretation on me. The initial intent is to centralize the validation, I think it's a good thing idea but at the expense of sending unnecessary data in some cases. 

The goal is not to add more work for everybody, I hereby declare this PR as unnecessary. Thank you for your patience ;)

Reply:
Oh I am sorry for not clarifying this earlier. I will close this PR 😄

Reply:
Actually, this change is still required as `p:XX!X` is less than 16, thus not valid. right? 👀 haha I'm sorry.